VAMobile/src/utils/inAppReviews.tsx

```diff
@@ -6,34 +6,70 @@ import { Events } from 'constants/analytics'
 
 import { logAnalyticsEvent } from './analytics'
 import { getVersionName } from './deviceData'
+import { useGiveFeedback } from './hooks'
 import { featureEnabled } from './remoteConfig'
 import { requestReview } from './rnReviews'
 
 export const STORAGE_REVIEW_EVENT_KEY = '@review_event'
+export const STORAGE_FEEDBACK_EVENT_KEY = '@feedback_event_'
 export const STORAGE_LAST_REVIEW_PROMPT_DATE_MILLIS = '@last_review_date'
+export const STORAGE_LAST_FEEDBACK_PROMPT_DATE_MILLIS = '@last_feedback_date'
 export const STORAGE_LAST_REVIEW_VERSION = '@last_review_version'
 export const IN_APP_REVIEW_INTERVAL_DAYS = 122
+export const IN_APP_FEEDBACK_INTERVAL_DAYS = 30
 export const IN_APP_REVIEW_ACTIONS_THRESHOLD = 7
+export const IN_APP_FEEDBACK_ACTIONS_THRESHOLD = 3
 
-export const registerReviewEvent = async (screenView?: boolean): Promise<void> => {
-  if (!featureEnabled('inAppReview')) return
-  const prev = await AsyncStorage.getItem(STORAGE_REVIEW_EVENT_KEY)
-  const total = prev ? parseInt(prev, 10) + 1 : 1
-  await AsyncStorage.setItem(STORAGE_REVIEW_EVENT_KEY, `${total}`)
-  const versionName = await getVersionName()
-  if (!screenView && total >= IN_APP_REVIEW_ACTIONS_THRESHOLD) {
-    const lastReview = await AsyncStorage.getItem(STORAGE_LAST_REVIEW_PROMPT_DATE_MILLIS)
-    if (!lastReview) {
-      await callReviewAPI(versionName)
-    } else {
-      const days = DateTime.fromMillis(parseInt(lastReview, 10)).diffNow('days').days
-      const lastReviewVersion = await AsyncStorage.getItem(STORAGE_LAST_REVIEW_VERSION)
-
-      if (days > IN_APP_REVIEW_INTERVAL_DAYS && lastReviewVersion !== versionName) {
+export const useReviewEvent = (screenView?: boolean, feedbackScreen?: string): (() => Promise<void>) => {
```

Comment:
You could move these args to the reviewEvent function that this function returns. Optional but it means fewer code changes overall. Also consider renaming screenView to isScreenView since it's a boolean